spec/components/layout_super_navigation_header_spec.rb

```diff
@@ -16,22 +16,73 @@ def component_name
 
     assert_select ".gem-c-layout-super-navigation-header__content[aria-labelledby]", count: 1
 
-    label = "##{assert_select('.gem-c-layout-super-navigation-header__content[aria-labelledby]').first['aria-labelledby']}"
+    label_id = "##{assert_select('.gem-c-layout-super-navigation-header__content[aria-labelledby]').first['aria-labelledby']}"
 
-    assert_select label, count: 1
+    assert_select label_id, count: 1
+  end
+
+  it "has `aria-labelledby` attributes correctly linked to elements that exist" do
+    render_component({})
+
+    label_ids = assert_select("[aria-labelledby]").map do |node|
+      node["aria-labelledby"]
+    end
+
+    label_ids.each do |label_id|
+      assert_select "##{label_id}", count: 1
+    end
+  end
+
+  it "has `aria-controls` attributes correctly linked to elements that exist" do
+    render_component({})
+
+    control_ids = assert_select("[aria-controls]").map do |button|
+      button["aria-controls"]
+    end
+
+    control_ids.each do |control_id|
+      assert_select "##{control_id}", count: 1
+    end
+  end
+
+  it "has buttons that have the three required attributes for the menu toggles to work" do
+    render_component({})
+
+    menu_buttons = assert_select("button[aria-controls][data-toggle-desktop-group][data-toggle-mobile-group]")
+    all_buttons = assert_select("button")
+
+    # There is a search submit button - so the corect number of menu buttons
```

Comment:
typo: correct

Reply:
Corrected in https://github.com/alphagov/govuk_publishing_components/pull/2223/commits/6e25269af674504c825bc2530c87014f3bea6042.